### PR TITLE
Feature/ms3 13 metadata extraction

### PR DESCRIPTION
**Pull request by rhoxhaj**
## Abstract
This endpoint will do: 
- Mesh will download the binary
- extract and store metadata in the same way like binary data do.

Any suggestion regarding code refactoring to use the same binary processors is welcomed.

INFO: There was a problem with the rebase. I will squash the commits before merging.

### CODE REVIEW

File: core/src/main/java/com/gentics/mesh/core/s3binary/AbstractS3BinaryProcessor.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.gentics.mesh.core.s3binary;
+
+
+/**
+ * @see BinaryDataProcessor
+ */
+public abstract class AbstractS3BinaryProcessor implements S3BinaryDataProcessor {
+// TODO empty abstract class is not needed. Remove this
```
Comment (plyhun) [inline core/src/main/java/com/gentics/mesh/core/s3binary/AbstractS3BinaryProcessor.java:8]: Dont forget about this TODO please :smiley:

**Review by plyhun — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/gentics/mesh/core/endpoint/node/S3BinaryMetadataExtractionHandlerImpl.java [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+									return fileName;
+								}
+
+								@Override
+								public String contentType() {
+									return mimeTypeForFilename;
+								}
+
+								@Override
+								public String contentTransferEncoding() {
```
Comment (plyhun) [inline core/src/main/java/com/gentics/mesh/core/endpoint/node/S3BinaryMetadataExtractionHandlerImpl.java:163]: Are these two nulls below correctly processed? 
If null is not expected there as a special case, I would place some existing (if possible) or default values there. Charset is usually UTF-8. 
If it is not possible to get these values at the moment, I will test this part explicitly during the final merge.

**Review by plyhun — COMMENTED** [review]
All look good, except the part with two dangling NULLs.

File: core/src/main/java/com/gentics/mesh/core/endpoint/node/S3BinaryMetadataExtractionHandlerImpl.java (hunk 2 above)
Comment (rhoxhaj) [reply core/src/main/java/com/gentics/mesh/core/endpoint/node/S3BinaryMetadataExtractionHandlerImpl.java:163]: I fixed it for the charset, but I cannot find any default value for the contentTransferEncoding. In any case I am not using these values at all.

**Pull request merged**